sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -292,6 +292,37 @@ class DataSourceV2SQLSuiteV1Filter
     }
   }
 
+  test("create v2 table with collation column") {
```

Comment:
I'd prefer to centralize the string collation tests. Can we extend `DatasourceV2SQLBase` in `CollationSuite` so that we can test v2 tables there?

Reply:
sure, no problem

Reply:
extending `DatasourceV2SQLBase` worked without issues in my IDE using maven, but has a very weird failure when using sbt which is why the PR check keeps failing. Is this a known thing perhaps?

`[info] - add collated column with alter table *** FAILED *** (237 milliseconds)
[info]   org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 140.0 failed 1 times, most recent failure: Lost task 0.0 in stage 140.0 (TID 134) (192.168.15.240 executor driver): java.lang.AssertionError: index (1) should < 1
[info]  at org.apache.spark.sql.catalyst.expressions.UnsafeRow.assertIndexIsValid(UnsafeRow.java:120)`

cc: @dbatomic